app/models/institution.rb

```diff
@@ -228,10 +228,16 @@ def self.autocomplete(search_term, limit = 6)
       .limit(limit)
   end
 
+  # This is used both on Weams imports and in the search scope
+  # Idea is to have a processed version of the institution column available to compare with
+  # the trigram % operator against the processed search term
   def self.institution_search_term(search_term)
+    return if search_term.blank?
+
     processed_search_term = search_term.clone
     Settings.search.common_word_list.each do |word|
-      processed_search_term = processed_search_term.gsub(Regexp.new(word, Regexp::IGNORECASE), '')
+      processed_search_term = processed_search_term.gsub(/\b#{Regexp.escape(word)}\b/i, '') if word.match(/[a-z]/i)
+      processed_search_term = processed_search_term.gsub(/#{Regexp.escape(word)}/i, '')
```

Comment:
these could be `gsub!` instead of assigning the variable each time

Reply:
changed, had to switch `clone` to `dup` to use